src/dali_executor/dali_pipeline.h

```diff
@@ -68,14 +69,18 @@ class DaliPipeline {
   ~DaliPipeline() {
     ReleasePipeline();
     ReleaseStream();
+    if (release_buffers_on_delete_) {
+      ReleaseBuffers();
+    }
   }
 
   DaliPipeline(const std::string& serialized_pipeline, int max_batch_size, int num_threads,
-               int device_id) :
+               int device_id, bool release_buffers_on_delete) :
```

Comment:
In this form, I foresee problems with poor allocation performance and fighting between pipelines. To name a few:
1. What happens if there are multiple pipelines with this flag? I think we should count them.
2. What happens if there are pipelines with and without this flag?

I'd suggest having a counter as a static member or a global variable and increment it when we create a new pipeline and decrement when we destroy it. Once all pipelines are down, we can call ReleaseUnusedBuffers, but not before.
If there's a mix of pipelines with and without the flag, we have no way of separating their memory pools, so we're looking at two imperfect options:
a) defer the memory release until all pipelines are destroyed - possibly keeping large memory utilization during upcoming requests
b) release immediately, which will leave the memory used by the pipelines without the flag (because it's still used), but may degrade their performance in upcoming requests.

Reply:
Thank you @mzient for very good points.

Regarding problem 1, Triton solves the case. It's not possible to unload subset of instances of a model - either you unload all or none. Nevertheless, I'll add the counter, as a sanity check.

Regarding problem 2, this is a backend option - it's set once per Triton process invocation. User sets it as I presented in description above:
```bash
tritonserver --backend-config=dali,release_after_unload=true
```
This way either all model instances have the option, or none of them. Therefore the situation you described (pipelines with and without the flag) is not possible. Since there's no perfect solution for this problem, I'll refrain from introducing a path for this. Is that OK for you?